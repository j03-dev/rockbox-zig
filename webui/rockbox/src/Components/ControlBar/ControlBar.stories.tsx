import type { Meta, StoryObj } from "@storybook/react";

import ControlBar from "./ControlBar";
import { fn } from "@storybook/test";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "Components/ControlBar",
  component: ControlBar,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {
    onPlay: { action: "onPlay" },
    onPause: { action: "onPause" },
    onNext: { action: "onNext" },
    onPrevious: { action: "onPrevious" },
    onShuffle: { action: "onShuffle" },
    onRepeat: { action: "onRepeat" },
  },
} satisfies Meta<typeof ControlBar>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Default: Story = {
  args: {
    onPlay: fn(),
    onPause: fn(),
    onNext: fn(),
    onPrevious: fn(),
    onShuffle: fn(),
    onRepeat: fn(),
  },
};

export const Playing: Story = {
  args: {
    nowPlaying: {
      album: "Forgotten Shapes",
      artist: "Waveshaper",
      title: "Disco on the Baltic Sea",
      cover:
        "https://resources.tidal.com/images/5b33d07f/7d28/417d/8f8f/86291d0b6b34/320x320.jpg",
      duration: 255488.00659179688,
      progress: 123456.789,
      isPlaying: true,
      albumId: "229251493",
    },
    onPlay: fn(),
    onPause: fn(),
    onNext: fn(),
    onPrevious: fn(),
    onShuffle: fn(),
    onRepeat: fn(),
  },
};
