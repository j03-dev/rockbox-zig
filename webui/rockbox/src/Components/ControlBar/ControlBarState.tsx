import { atom } from "recoil";
import { CurrentTrack, Track } from "../../Types/track";

export const controlBarState = atom<{
  nowPlaying?: CurrentTrack;
  locked?: boolean;
  previousTracks?: Track[];
  nextTracks?: Track[];
}>({
  key: "controlBarState",
  default: {
    nowPlaying: undefined,
    locked: false,
    previousTracks: [],
    nextTracks: [],
  },
});
