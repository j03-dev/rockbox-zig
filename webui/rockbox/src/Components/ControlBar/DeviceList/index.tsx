import DeviceListWithData from "./DeviceListWithData";

export default DeviceListWithData;
