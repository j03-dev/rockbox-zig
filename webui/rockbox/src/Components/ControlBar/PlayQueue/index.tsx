import PlayQueueWithData from "./PlayQueueWithData";

export default PlayQueueWithData;
