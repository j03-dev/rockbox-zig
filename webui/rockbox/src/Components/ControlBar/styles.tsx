import styled from "@emotion/styled";

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  height: 60px;
  margin-top: 5px;
  margin-bottom: 20px;
`;

export const ControlsContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
  width: 160px;
`;

export const Button = styled.button`
  background-color: transparent;
  cursor: pointer;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  &:hover {
    opacity: 0.6;
  }
`;

export const Controls = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex: 0.3;
`;
