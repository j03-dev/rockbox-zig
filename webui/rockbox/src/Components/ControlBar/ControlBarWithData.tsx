import { FC } from "react";
import ControlBar from "./ControlBar";

const ControlBarWithData: FC = () => {
  return (
    <ControlBar
      nowPlaying={{
        album: "Pluto x Baby Pluto (Deluxe)",
        artist: "Future, Lil Uzi Vert",
        title: "Drankin N Smokin",
        cover:
          "https://resources.tidal.com/images/fe6787d5/4ba5/4d3e/8576/48943ee6a768/320x320.jpg",
        duration: 255488.00659179688,
        progress: 123456.789,
        isPlaying: true,
        albumId: "229251493",
      }}
      onPlay={() => {}}
      onPause={() => {}}
      onNext={() => {}}
      onPrevious={() => {}}
      onShuffle={() => {}}
      onRepeat={() => {}}
    />
  );
};

export default ControlBarWithData;
