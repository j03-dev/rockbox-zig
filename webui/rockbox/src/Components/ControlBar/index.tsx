import ControlBar from "./ControlBarWithData";

export default ControlBar;
