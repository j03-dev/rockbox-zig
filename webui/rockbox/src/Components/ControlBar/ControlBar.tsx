import { FC } from "react";
import Play from "../Icons/Play";
import Previous from "../Icons/Previous";
import Next from "../Icons/Next";
import Shuffle from "../Icons/Shuffle";
import Repeat from "../Icons/Repeat";
import { Button, Container, Controls, ControlsContainer } from "./styles";
import CurrentTrack from "./CurrentTrack";
import RightMenu from "./RightMenu";
import Pause from "../Icons/Pause";
import { CurrentTrack as NowPlaying } from "../../Types/track";

export type ControlBarProps = {
  nowPlaying?: NowPlaying;
  onPlay: () => Promise<void>;
  onPause: () => void;
  onNext: () => void;
  onPrevious: () => void;
  onShuffle: () => void;
  onRepeat: () => void;
  liked?: boolean;
  onLike: (trackId: string) => void;
  onUnlike: (trackId: string) => void;
};

const ControlBar: FC<ControlBarProps> = (props) => {
  return (
    <Container>
      <Controls>
        <ControlsContainer>
          <Button onClick={props.onShuffle}>
            <Shuffle />
          </Button>
          <Button onClick={props.onPrevious}>
            <Previous />
          </Button>
          {!props.nowPlaying?.isPlaying && (
            <Button onClick={props.onPlay}>
              <Play />
            </Button>
          )}
          {props.nowPlaying?.isPlaying && (
            <Button onClick={props.onPause}>
              <Pause />
            </Button>
          )}
          <Button onClick={props.onNext}>
            <Next />
          </Button>
          <Button onClick={props.onRepeat}>
            <Repeat />
          </Button>
        </ControlsContainer>
      </Controls>
      <CurrentTrack
        nowPlaying={props.nowPlaying}
        liked={props.liked}
        onLike={props.onLike}
        onUnlike={props.onUnlike}
      />
      <RightMenu />
    </Container>
  );
};

export default ControlBar;
