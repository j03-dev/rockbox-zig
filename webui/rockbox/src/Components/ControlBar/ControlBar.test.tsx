import { render } from "@testing-library/react";
import { vi } from "vitest";
import ControlBar from "./ControlBar";
import Providers from "../../Providers";

describe("ControlBar", () => {
  it("should render", () => {
    const { container } = render(
      <Providers>
        <ControlBar
          nowPlaying={{
            album: "Forgotten Shapes",
            artist: "Waveshaper",
            title: "Disco on the Baltic Sea",
            cover:
              "https://resources.tidal.com/images/5b33d07f/7d28/417d/8f8f/86291d0b6b34/320x320.jpg",
            duration: 255488.00659179688,
            progress: 123456.789,
            isPlaying: true,
            albumId: "229251493",
          }}
          onPlay={vi.fn()}
          onPause={vi.fn()}
          onNext={vi.fn()}
          onPrevious={vi.fn()}
          onShuffle={vi.fn()}
          onRepeat={vi.fn()}
        />
      </Providers>
    );
    expect(container).toMatchSnapshot();
  });
});
