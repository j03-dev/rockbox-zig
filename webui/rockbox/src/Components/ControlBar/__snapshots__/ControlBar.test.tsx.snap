// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ControlBar > should render 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="css-1dlrxtg"
    >
      <div
        class="css-1h69jy4"
      >
        <div
          class="css-1ytum11"
        >
          <button
            class="css-10asy2d"
          >
            <svg
              data-test="icon-PlaybackShuffle"
              fill="none"
              height="16"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.2 7h2.85l-1.3 1.3c-.4.4-.4 1 0 1.4.25.2.5.3.75.3s.5-.1.7-.3l3-3 .15-.15.05-.05c0-.025.013-.037.025-.05.012-.013.025-.025.025-.05 0-.021 0-.033.004-.044.005-.015.017-.027.046-.056 0-.021 0-.033.004-.044.005-.015.017-.027.046-.056.05-.15.05-.25 0-.4 0-.05 0-.05-.05-.1 0-.021 0-.033-.004-.044-.005-.015-.017-.027-.046-.056 0-.025-.012-.037-.025-.05-.012-.013-.025-.025-.025-.05 0 0 0-.05-.05-.05l-.15-.15-3-3c-.4-.4-1-.4-1.4 0-.4.4-.4 1 0 1.4L19.1 5h-2.85c-1.75 0-3.3.9-4.25 2.35L6.85 15.6c-.55.9-1.5 1.4-2.55 1.4H2.5c-.55 0-1 .45-1 1s.45 1 1 1h1.8c1.75 0 3.3-.9 4.25-2.35L13.7 8.4c.5-.85 1.5-1.4 2.5-1.4ZM2.5 7h1.8c1.05 0 2 .5 2.55 1.4l.45.75 1.2-1.9C7.55 5.8 6 4.95 4.3 4.95H2.5c-.55 0-1 .45-1 1 0 .6.45 1.05 1 1.05Zm19.95 11.2v.05c0 .05 0 .1-.05.1v.05c-.05.1-.1.2-.2.3l-3 3c-.2.2-.45.3-.7.3-.25 0-.5-.1-.75-.3-.4-.4-.4-1 0-1.4l1.3-1.3H16.2c-1.7 0-3.25-.85-4.2-2.3l1.25-1.9.45.75c.55.9 1.5 1.4 2.55 1.4h2.85l-1.3-1.3c-.4-.4-.4-1 0-1.4.4-.4 1-.4 1.4 0l3 3c.1.1.15.2.2.3v.05c.05.05.05.1.05.15v.05c.05.15.05.25 0 .4Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <button
            class="css-10asy2d"
          >
            <svg
              fill="none"
              height="18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="ionicon"
                d="M3.55 2c.304 0 .55.243.55.542V7.16l8.001-4.716a1.224 1.224 0 0 1 1.23-.015c.413.23.669.677.669 1.162v9.818c0 .486-.256.932-.669 1.162-.382.216-.853.21-1.23-.015L4.1 9.84v4.618a.546.546 0 0 1-.55.542.546.546 0 0 1-.55-.542V2.542C3 2.243 3.246 2 3.55 2Z"
                style="fill: #000;"
              />
            </svg>
          </button>
          <button
            class="css-10asy2d"
          >
            <svg
              height="32"
              viewBox="0 0 512 512"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Pause
              </title>
              <path
                d="M208 432h-48a16 16 0 0 1-16-16V96a16 16 0 0 1 16-16h48a16 16 0 0 1 16 16v320a16 16 0 0 1-16 16zm144 0h-48a16 16 0 0 1-16-16V96a16 16 0 0 1 16-16h48a16 16 0 0 1 16 16v320a16 16 0 0 1-16 16z"
                style="fill: #000;"
              />
            </svg>
          </button>
          <button
            class="css-10asy2d"
          >
            <svg
              fill="none"
              height="18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="ionicon"
                d="M14.45 2a.546.546 0 0 0-.55.542V7.16L5.899 2.444a1.224 1.224 0 0 0-1.23-.015C4.256 2.659 4 3.105 4 3.591v9.818c0 .486.256.932.669 1.162.382.216.853.21 1.23-.015L13.9 9.84v4.618c0 .299.246.542.55.542.304 0 .55-.243.55-.542V2.542A.546.546 0 0 0 14.45 2Z"
                style="fill: #000;"
              />
            </svg>
          </button>
          <button
            class="css-10asy2d"
          >
            <svg
              data-test="icon-PlaybackInactiveRepeat"
              fill="none"
              height="16"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.293 1.293a1 1 0 0 1 1.414 0l3 3a1 1 0 0 1 0 1.414l-3 3a1 1 0 1 1-1.414-1.414L17.586 6H8.5c-3.298 0-6 2.702-6 6a1 1 0 1 1-2 0c0-4.402 3.598-8 8-8h9.086l-1.293-1.293a1 1 0 0 1 0-1.414ZM23.5 12a1 1 0 1 0-2 0c0 3.298-2.702 6-6 6H6.414l1.293-1.293a1 1 0 1 0-1.414-1.414l-3 3a1 1 0 0 0 0 1.414l3 3a1 1 0 0 0 1.414-1.414L6.414 20H15.5c4.402 0 8-3.598 8-8Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-frud38"
      >
        <img
          alt="Album cover"
          class="css-31cka"
          src="https://resources.tidal.com/images/5b33d07f/7d28/417d/8f8f/86291d0b6b34/320x320.jpg"
        />
        <div
          class="css-rbpzuh"
        >
          <div
            class="css-1cojt9j"
          >
            Disco on the Baltic Sea
          </div>
          <div
            style="display: flex; flex-direction: row; align-items: center; justify-content: space-between;"
          >
            <div
              class="css-1jnbdma"
            >
              02:03
            </div>
            <div
              class="css-bdl275"
            >
              Waveshaper
              <span
                class="css-c9cgcr"
              >
                -
              </span>
              Forgotten Shapes
            </div>
            <div
              class="css-1jnbdma"
            >
              04:15
            </div>
          </div>
          <div
            class="css-1n7uud4"
          >
            <div
              aria-label="48% Loaded"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="48.3219508605943"
              class="ae"
              data-baseweb="progress-bar"
              role="progressbar"
            >
              <div
                class="af ag ah ai aj"
              >
                <div
                  class="ak al am an ao ap aq ar"
                >
                  <div
                    class="ak al am an as at ae au av"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-nb4c1y"
      >
        <button
          class="css-10asy2d"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 dmvaRK"
            fill="currentColor"
            focusable="false"
            height="21"
            viewBox="0 0 20 20"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.4 9H8.6c-.552 0-.6.447-.6 1s.048 1 .6 1h5.8c.552 0 .6-.447.6-1s-.048-1-.6-1zm2 5H8.6c-.552 0-.6.447-.6 1s.048 1 .6 1h7.8c.552 0 .6-.447.6-1s-.048-1-.6-1zM8.6 6h7.8c.552 0 .6-.447.6-1s-.048-1-.6-1H8.6c-.552 0-.6.447-.6 1s.048 1 .6 1zM5.4 9H3.6c-.552 0-.6.447-.6 1s.048 1 .6 1h1.8c.552 0 .6-.447.6-1s-.048-1-.6-1zm0 5H3.6c-.552 0-.6.447-.6 1s.048 1 .6 1h1.8c.552 0 .6-.447.6-1s-.048-1-.6-1zm0-10H3.6c-.552 0-.6.447-.6 1s.048 1 .6 1h1.8c.552 0 .6-.447.6-1s-.048-1-.6-1z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    class=""
  />
</div>
`;
